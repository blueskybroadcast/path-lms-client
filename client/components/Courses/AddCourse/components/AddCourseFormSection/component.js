import React from 'react';
import PropTypes from 'prop-types';

const AddCourseFormSection = ({
  label, isRequired, children, tooltip
}) => (
  <fieldset>
    <legend>
      <span>{label}</span>
    </legend>
    <div className="control-group">
      <label htmlFor={label}>
        {label}
        <span className="has-tooltip">
          { tooltip && (
            <span
              className="tooltiped"
              data-toggle="tooltip"
              data-original-title={tooltip}
            >
              <i className="icon icon-question-circle" />
            </span>
          )}
        </span>
        { isRequired && <abbr title="required" className="required">*</abbr> }
      </label>
      {children}
    </div>
  </fieldset>
);

AddCourseFormSection.propTypes = {
  label: PropTypes.string.isRequired,
  isRequired: PropTypes.bool,
  children: PropTypes.node.isRequired,
  tooltip: PropTypes.string
};

AddCourseFormSection.defaultProps = {
  isRequired: false,
  tooltip: null
};

export default AddCourseFormSection;
