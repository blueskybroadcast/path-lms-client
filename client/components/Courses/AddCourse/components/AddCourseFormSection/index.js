import AddCourseFormSection from './component';

export default AddCourseFormSection;
