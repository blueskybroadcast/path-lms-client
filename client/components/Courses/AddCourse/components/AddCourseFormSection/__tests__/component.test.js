import React from 'react';
import { shallow } from 'enzyme';
import AddCourseFormSection from '../component';

describe('AddCourseFormSection', () => {
  test('should render AddCourseFormSection with all props correctly', () => {
    const wrapper = shallow(
      <AddCourseFormSection
        label="Label"
        isRequired
        tooltip="Tooltip text"
      >
        <div>Children</div>
      </AddCourseFormSection>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
