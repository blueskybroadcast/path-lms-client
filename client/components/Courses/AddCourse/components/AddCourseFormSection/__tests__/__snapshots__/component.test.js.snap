// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseFormSection should render AddCourseFormSection with all props correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Label
    </span>
  </legend>
  <div
    className="control-group"
  >
    <label
      htmlFor="Label"
    >
      Label
      <span
        className="has-tooltip"
      >
        <span
          className="tooltiped"
          data-original-title="Tooltip text"
          data-toggle="tooltip"
        >
          <i
            className="icon icon-question-circle"
          />
        </span>
      </span>
      <abbr
        className="required"
        title="required"
      >
        *
      </abbr>
    </label>
    <div>
      Children
    </div>
  </div>
</fieldset>
`;
